--- crossfire-client-1.7.0.orig/gtk/image.c	2004-04-19 08:35:11.000000000 +0200
+++ crossfire-client-1.7.0/gtk/image.c	2004-04-30 09:33:58.000000000 +0200
@@ -193,11 +193,11 @@
 	}
 
 	for (i=0; i < pi->map_width * pi->map_height; i+= 4) {
-	    p = (uint32) (fog->pixels + i);
-	    g = ( ((p >> 24) & 0xff)  + ((p >> 16) & 0xff) + ((p >> 8) & 0xff)) / 3;
-	    p = (g << 24) | (g << 16) | (g << 8) | (p & 0xff);
+	    p = (uint32*) (fog->pixels + i);
+	    g = ( ((*p >> 24) & 0xff)  + ((*p >> 16) & 0xff) + ((*p >> 8) & 0xff)) / 3;
+	    p = (g << 24) | (g << 16) | (g << 8) | (*p & 0xff);
 	    l = (uint32*) fog->pixels + i;
-	    *(uint32*) l = p;
+	    *(uint32*) l = *p;
 	}
 
 	SDL_UnlockSurface(fog);
