--- crossfire-client-1.5.0/gtk/gx11.c.errno	Tue Feb 25 08:49:40 2003
+++ crossfire-client-1.5.0/gtk/gx11.c	Mon Mar 10 14:51:12 2003
@@ -64,7 +64,7 @@
 
 #include "config.h"
 
-#ifdef __CYGWIN__
+#ifdef HAVE_ERRNO_H
 #include <errno.h>
 #endif
 
--- crossfire-client-1.5.0/common/config.h.in.errno	Tue Feb 25 08:49:40 2003
+++ crossfire-client-1.5.0/common/config.h.in	Mon Mar 10 14:56:15 2003
@@ -66,6 +66,9 @@
 /* Define if you have the <unistd.h> header file.  */
 #undef HAVE_UNISTD_H
 
+/* Define if you have the <errno.h> header file.  */
+#undef HAVE_ERRNO_H
+
 /* Define if you have the m library (-lm).  */
 #undef HAVE_LIBM
 
--- crossfire-client-1.5.0/configure.in.errno	Mon Mar 10 14:52:40 2003
+++ crossfire-client-1.5.0/configure.in	Mon Mar 10 14:52:52 2003
@@ -321,7 +321,7 @@
 
 dnl Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS(fcntl.h sys/ioctl.h sys/time.h unistd.h string.h sys/select.h)
+AC_CHECK_HEADERS(fcntl.h sys/ioctl.h sys/time.h unistd.h string.h sys/select.h errno.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
