diff -durN crossfire-client-0.95.8.orig/configure.in crossfire-client-0.95.8/configure.in
--- crossfire-client-0.95.8.orig/configure.in	Wed Nov 29 06:52:21 2000
+++ crossfire-client-0.95.8/configure.in	Mon Jan 29 12:11:08 2001
@@ -144,7 +144,7 @@
     LDFLAGS="$LDFLAGS"
 fi
 
-AC_CHECK_LIB(asound, main, alsa_sound=yes, , -lm)
+dnl AC_CHECK_LIB(asound, main, alsa_sound=yes, , -lm)
 AC_CHECK_LIB(audio, alNewConfig, sgi_sound=yes, )
 AC_CHECK_HEADER(sys/soundcard.h, oss_sound=yes, oss_sound=no)
 AC_CHECK_HEADER(sys/audioio.h, sun_sound=yes, sun_sound=no)
