diff -u -r1.48 -r1.50
--- gtk/gx11.c	4 Apr 2004 10:07:47 -0000	1.48
+++ gtk/gx11.c	16 May 2004 06:09:21 -0000	1.50
@@ -677,15 +677,39 @@
 
 
 /* Event handlers for map drawing area */
-
+/* For a reason I don't know, this gets called a whole bunch.
+ * every time a monster is killed, this gets called for a reason
+ * I can't figure out.
+ */
 
 static gint
 configure_event (GtkWidget *widget, GdkEventConfigure *event)
 {
+    static sint16  ox=-1, oy=-1;
+
+    /* Handle the surplus number of events that this causes to be generated.
+     * basically, if the size of the window hasn't changed, we really don't
+     * care - position of the window isn't important.
+     * note that we could be more clever and free up the other data even on
+     * requests that do change the size,
+     * but this will fix the most horrendous memory leak
+     */
+    if (event->type == GDK_CONFIGURE) {
+	if (((GdkEventConfigure*)event)->width == ox && ((GdkEventConfigure*)event)->height == oy)
+	    return TRUE;
+	else {
+#if 0
+	    fprintf(stderr, "ox=%d != %d, oy=%d != %d\n", ox, ((GdkEventConfigure*)event)->width,
+		    oy, ((GdkEventConfigure*)event)->height);
+#endif
+	    ox = ((GdkEventConfigure*)event)->width;
+	    oy = ((GdkEventConfigure*)event)->height;
+	}
+    }
 
 #ifdef HAVE_SDL
     if(use_config[CONFIG_SDL]) {
-	/* When program first runs, mapsruface can be null.
+	/* When program first runs, mapsurface can be null.
 	 * either way, we want to catch it here.
 	 */
 	if (mapsurface)
