diff -u -r1.17 -r1.20
--- gtk/image.c	30 Jan 2004 18:43:34 -0000	1.17
+++ gtk/image.c	22 May 2004 15:51:50 -0000	1.20
@@ -221,10 +235,10 @@
     if (pi->map_image) {
 #ifdef HAVE_SDL
 	if (use_config[CONFIG_SDL]) {
-	    SDL_FreeSurface(pi->map_image);
 	    free(((SDL_Surface*)pi->map_image)->pixels);
+	    SDL_FreeSurface(pi->map_image);
 	    SDL_FreeSurface(pi->fog_image);
-	    free(((SDL_Surface*)pi->fog_image)->pixels);
+/*	    free(((SDL_Surface*)pi->fog_image)->pixels);*/
 	}
 	else
 #endif
