--- crossfire-client-1.60.0/gtk-v2/src/png.c.wiget	2011-01-28 07:17:57.000000000 +0100
+++ crossfire-client-1.60.0/gtk-v2/src/png.c	2012-02-18 17:52:15.410954778 +0100
@@ -571,7 +571,7 @@ int png_to_gdkpixmap(GdkWindow *window,
         png_destroy_read_struct (&png_ptr, NULL, NULL);
         return PNGX_OUTOFMEM;
     }
-    if (setjmp (png_ptr->jmpbuf)) {
+    if (setjmp (png_jmpbuf(png_ptr))) {
         png_destroy_read_struct (&png_ptr, &info_ptr,NULL);
         return PNGX_DATA;
     }
