--- crossfire-client-1.3.0/configure.in.wiget	Fri Jul 12 15:29:44 2002
+++ crossfire-client-1.3.0/configure.in	Fri Jul 12 15:32:50 2002
@@ -19,8 +19,6 @@
 
 AC_CONFIG_HEADER(common/config.h)
 
-AM_ACLOCAL_INCLUDE(macros)
-
 dnl Check for some neeed programs
 AC_PATH_PROG(MKDIR, mkdir)
 AC_PATH_PROG(TAR, tar)
@@ -228,7 +226,7 @@
   AM_PATH_SDL(1.1.3)
   if eval "test x$no_sdl = x"; then
       AC_CHECK_LIB( SDL_image, IMG_LoadPNG_RW, 
-                    have_sdlimage="yes", have_sdlimage="no", $SDL_CFLAGS) 
+                    have_sdlimage="yes", have_sdlimage="no",[$SDL_CFLAGS $SDL_LIBS]) 
       if eval "test x$have_sdlimage = xyes"; then
           SDL_LIBS="$SDL_LIBS -lSDL_image"
       else
